### ISSUE TITLE
Multiple response URLs not working correctly

### ISSUE THREAD

**Issue by ngaya-ll** [getsentry/responses#170]
The fix for #161 breaks multi-URL functionality.

Test case:
```
def test_multiple_urls():
    @responses.activate
    def run():
        responses.add(responses.GET, 'http://example.com/one', body='one')
        responses.add(responses.GET, 'http://example.com/two', body='two')

        resp = requests.get('http://example.com/two')
        assert_response(resp, 'two')
        resp = requests.get('http://example.com/one')
        assert_response(resp, 'one')

    run()
    assert_reset()
```

**Comment by dcramer** [comment]
gonna get the regression test in ahead of a fix

**Comment by ngaya-ll** [comment]
I have a fix, will post in next 5 min.

**Comment by dcramer** [comment]
:+1: thanks!

**Comment by bartekwojcicki** [comment]
I'm sorry for introducing this one, thank you both for the fix.

**Comment by fauskanger** [comment]
[ex]: https://github.com/getsentry/responses#multiple-responses

I've used the [Multiple responses example][ex], and it works:
    
    @responses.activate
    def test_multiple_to_same_url():
        responses.add(responses.GET, 'http://twitter.com/api/1/foobar', status=500)
        responses.add(responses.GET, 'http://twitter.com/api/1/foobar',
                      body='{}', status=200,
                      content_type='application/json')
    
        resp = requests.get('http://twitter.com/api/1/foobar')
        assert resp.status_code == 500
        resp = requests.get('http://twitter.com/api/1/foobar')
        assert resp.status_code == 200

But if I rearrange so that the a request has been made to _the same_ url, between adding the response it doesn't work.


    @responses.activate
    def test_multiple_to_same_url():
        responses.add(responses.GET, 'http://twitter.com/api/1/foobar', status=500)
    
        resp = requests.get('http://twitter.com/api/1/foobar')
        assert resp.status_code == 500
    
    
        responses.add(responses.GET, 'http://twitter.com/api/1/foobar',
                      body='{}', status=200,
                      content_type='application/json')
        resp = requests.get('http://twitter.com/api/1/foobar')
        assert resp.status_code == 200

The error output:

    ================================== FAILURES ===================================
    __________________________ test_multiple_to_same_url __________________________
    
        @responses.activate
        def test_multiple_to_same_url():
            responses.add(responses.GET, 'http://twitter.com/api/1/foobar', status=500)
    
            resp = requests.get('http://twitter.com/api/1/foobar')
            assert resp.status_code == 500
    
    
            responses.add(responses.GET, 'http://twitter.com/api/1/foobar',
                          body='{}', status=200,
                          content_type='application/json')
            resp = requests.get('http://twitter.com/api/1/foobar')
    >       assert resp.status_code == 200
    E       assert 500 == 200
    E        +  where 500 = <Response [500]>.status_code

Adding responses and make requests to _other_ urls works fine too. In other words, it seems that __multiple responses to the same url must be added before requests to them are made__, but other responses can independently be added and be called.

**Comment by dcramer** [comment]
@fauskanger IMO it should work as you'd expect there, but given I haven't looked at the code in a few weeks its not obvious to me why it wouldn't be.

**Comment by ngaya-ll** [comment]
This is behaving as expected per https://github.com/getsentry/responses/pull/171.

* Initially there is just one response registered.
* At the time of the first request is made, there is only one matching response so it is not discarded.
* When the second response is added, it is appended to the list of responses.
* When the second request is made, it iterates over the list and returns the first response again.  At this point the first response is discarded.
* Subsequent requests will return the second response.

**Comment by fauskanger** [comment]
Thanks for reply back and for the explanation. I added my comment for others to read, as it wasn't clear to me.

In my case I want to test an API for all errors, let's say __n__ different cases. 
I had __n__ pairs of interleaved 

    response.add(...) 
    api_lib.call_comething()

before I understood that I had to declare the endpoint url with different data __n__ times before I made the first request.

When this didn't work I was worried that it (for some obscure reason) would be incompatible with other tools I used. Anyway, now that I made it work, __I'd like to thank all contributors for this excellent tool__! 

(The last two days I've used ``requests_mock``, ``HTTPretty``, ``python-mock`` and this. I've landed on ``responses``.)
